packages/stockflux-launcher/src/App.js

```diff
@@ -1,69 +1,112 @@
+  const defaultWidth = options ? options.defaultWidth : 50;
+  const isDockable = options ? options.customData.isDockable : false;
+  const [leftPosition, setLeft] = useState(0);
+  const [topPosition, setTop] = useState(0);
+  const [isHorizontal, setHorizontal] = useState(true);
   const [edge, windowActions] = useDockWindow(
     ScreenEdge.TOP,
     OpenfinApiHelpers.getCurrentWindowSync(),
     true,
-    { dockedWidth: 50, dockedHeight: 50 }
+    {
+      dockedWidth: isHorizontal ? defaultWidth : 80,
+      dockedHeight: defaultHeight
+    },
+    {
+      undockPosition: { left: leftPosition, top: topPosition },
+      undockSize: { width: 724, height: 88 }
+    }
   );
 
-  const prevEdgeRef = useRef();
+  const iconStyle = isHorizontal ? 'icon horizontal' : 'icon vertical';
+  const toolbarStyle = isHorizontal ? 'toolbar' : 'toolbar t-vertical';
+
+  /* Hook undock if initialDocked is false on start */
+  useEffect(() => {
+    if (
+      options &&
+      options.customData &&
+      options.customData.initialDocked === false
+    ) {
+      windowActions.dockNone();
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [options]);
+
+  /* Make sure window shrinks when dragged from top */
   useEffect(() => {
-    prevEdgeRef.current = edge;
-  });
-  const prevEdge = prevEdgeRef.current;
+    if (edge === ScreenEdge.NONE) {
+      getUndockedPosition().then(position => {
+        setLeft(position.left);
+        setTop(position.top);
+
+        windowActions.dockNone();
+      });
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [edge]);
 
-  const edgeToBeChecked = edge === ScreenEdge.NONE ? prevEdge : edge;
+  useEffect(() => {
+    setHorizontal(
+      edge === ScreenEdge.TOP || edge === ScreenEdge.NONE ? true : false
+    );
+  }, [edge]);
 
   return (
-    <div className={cx('app', edgeToBeChecked)}>
-      {!isLauncherHorizontal(edgeToBeChecked) && CloseButton}
-      <AppShortcuts />
-      <FreeTextSearch
-        dockedTo={edge}
-        showTextInput={!isLauncherHorizontal(edgeToBeChecked)}
-      />
-      <ToolBar
-        tools={[
-          {
-            label: <FaChevronUp />,
-            onClick: windowActions.dockTop,
-            disabled: edge === ScreenEdge.TOP
-          },
-          {
-            label: <FaChevronLeft />,
-            onClick: windowActions.dockLeft,
-            disabled: edge === ScreenEdge.LEFT
-          },
-          {
-            label: <FaChevronRight />,
-            onClick: windowActions.dockRight,
-            disabled: edge === ScreenEdge.RIGHT
-          },
-          {
-            label: <FaRegHandRock />,
-            className: 'drag-handle'
-          }
-        ]}
-      />
-      {isLauncherHorizontal(edgeToBeChecked) && CloseButton}
+    <div className="launcher-container">
+      <div className="launcher-title">
+        <Titlebar dockedTo={edge} />
+      </div>
+      <div className={cx('app', edge)}>
```

Comment:
Is using `cx` here is necessary?

Reply:
Needed to evaluate to `app top`, `app none`, which share styles, or `app left` or `app right`, which share styles